data/org.freedesktop.portal.Usb.xml

```diff
@@ -0,0 +1,228 @@
+<?xml version="1.0"?>
+<!--
+ Copyright (C) 2020 Endless OS Foundation LLC
+               2023 GNOME Foundation Inc.
+
+ This library is free software; you can redistribute it and/or
+ modify it under the terms of the GNU Lesser General Public
+ License as published by the Free Software Foundation; either
+ version 2 of the License, or (at your option) any later version.
+
+ This library is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ Lesser General Public License for more details.
+
+ You should have received a copy of the GNU Lesser General Public
+ License along with this library. If not, see <http://www.gnu.org/licenses/>.
+
+ Author: Georges Basile Stavracas Neto <georges.stavracas@gmail.com>
+         Ryan Gonzalez <rymg19+github@gmail.com>
+-->
+
+<node name="/" xmlns:doc="http://www.freedesktop.org/dbus/1.0/doc.dtd">
+  <!--
+      org.freedesktop.portal.Usb:
+      @short_description: Portal for USB device access
+
+      This interface lets sandboxed applications monitor and request
+      access to connected USB devices.
+
+      Applications should prefer specialized portals for specific
+      device types, such as the Camera portal for cameras.
+
+      This documentation describes version 1 of this interface.
+    -->
+  <interface name="org.freedesktop.portal.Usb">
+
+    <!--
+        CreateSession:
+        @options: Vardict with optional further information
+        @session_handle: Object path for the session.
+
+        Creates an USB monitoring session. This is only necessary to
+        receive device events, like device being plugged or unplugged.
+
+        Supported keys in the @options vardict include:
+
+        * ``session_handle_token`` (``s``)
+
+          A string that will be used as the last element of the session handle. Must be a valid
+          object path element. See the #org.freedesktop.portal.Session documentation for
+          more information about the session handle.
+    -->
+    <method name="CreateSession">
+      <arg type="a{sv}" name="options" direction="in"/>
+      <arg type="o" name="session_handle" direction="out"/>
+    </method>
+
+    <!--
+        EnumerateDevices:
+        @options: Vardict with optional further information. There is currently
+        no option supported.
+
+        Enumerates all connected USB devices that this application has permission
+        to see.
+
+        The following results are returned in the @device vardict:
+
+        * ``id`` (``s``)
+
+          The portal-specific unique identifier of the device.
+
+        * ``parent`` (``s``)
+
+          Device ID of the parent device.
+
+        * ``readable`` (``b``)
```

Comment:
What use is a device which can not be read from? `AcquireDevices` also doesn't have a `readable` only a `writable`. Pretty sure this should just be removed.

Reply:
Until very recently it would have listed the devices even if it couldn't access them. It is devices that can be enumerated after all.